DiscordAlertsTrader/brokerages/ibkr_api.py

```diff
@@ -0,0 +1,525 @@
+            contract = Option(symbol=symbol_kwargs['symbol'], lastTradeDateOrContractMonth=symbol_kwargs['lastTradeDateOrContractMonth'], \
+                              strike=symbol_kwargs['strike'], right=symbol_kwargs['right'], \
+                                exchange='SMART', currency='USD')
+            contracts = self.ib.qualifyContracts(contract)
+            contract = contracts[0] if contracts else None
+            conId = contract.conId if contract else None
+            return conId
+        elif " " in symbol:
+            symbol_kwargs = self._convert_option_fromts(symbol)
+            contract = Option(symbol=symbol_kwargs['symbol'], lastTradeDateOrContractMonth=symbol_kwargs['lastTradeDateOrContractMonth'], \
+                              strike=symbol_kwargs['strike'], right=symbol_kwargs['right'], \
+                                exchange='SMART', currency='USD')
+            contracts = self.ib.qualifyContracts(contract)
+            contract = contracts[0] if contracts else None
+            conId = contract.conId if contract else None
+            return conId
+        else:
+            contracts = self.ib.qualifyContracts(Stock(symbol, 'SMART', 'USD'))
+            contract = contracts[0] if contracts else None
+            conId = contract.conId if contract else None
+            return conId
+    
+    def make_BTO_lim_order(self, Symbol:str, Qty:int, price:float, action="BTO", **kwarg):
+        "Buy with a limit order"
+
+        kwargs = {}
+        if action == "BTO":
+            kwargs['action'] = "BUY"
+        elif action == "STO":
+            kwargs['action'] = "SELL"
+        Symbol = self.fix_symbol(Symbol, "in")
+
+        if (("_" in Symbol) or (" " in Symbol)):
+            kwargs['asset'] = 'OPT'
+
+        else:
+            kwargs['asset'] ='STK'
+            kwargs['outsideRegularTradingHour'] = True
+            kwargs['stock'] = Symbol
+            
+        kwargs['enforce'] ='GTC'
+        kwargs['quant'] = Qty  
+        kwargs['orderType'] = 'LMT' 
+        kwargs['lmtPrice'] = price
+        kwargs['conId'] = self.get_con_id(Symbol)
+
+        return kwargs if kwargs['conId'] is not None else None
+
+    def make_Lim_SL_order(self, Symbol:str, Qty:int,  PT:float, SL:None, action="STC", **kwarg):
+        """Sell with a limit order and a stop loss order"""        
+        kwargs = {}
+        if action == "STC":
+            kwargs['action'] = "SELL"
+        elif action == "BTC":
+            kwargs['action'] = "BUY"
+
+        Symbol = self.fix_symbol(Symbol, "in")
+        if (("_" in Symbol) or (" " in Symbol)):
+            kwargs['asset'] = 'OPT'
+
+        else:
+            kwargs['asset'] ='STK'
+            kwargs['outsideRegularTradingHour'] = True
+            kwargs['stock'] = Symbol
+
+        kwargs['enforce'] ='GTC'
+        kwargs['quant'] = Qty 
+        kwargs['conId'] = self.get_con_id(Symbol)
+        if SL is not None:
```

Comment:
IBKR allows bracket orders. That function should create a bracket order, you are not doing that

Reply:
Need three parameters for entry, Take Profit and Sop Loss prices but only two are given, should add the entry price parameter? https://ib-insync.readthedocs.io/api.html#ib_insync.ib.IB.bracketOrder

Reply:
Do you mean this? https://ib-insync.readthedocs.io/api.html#ib_insync.ib.IB.oneCancelsAll